azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -195,87 +213,94 @@ def _fetch_goal_state(wire_client):
 
         wire_client.update_host_plugin(container_id, role_config_name)
 
-        return xml_text, xml_doc, incarnation
+        return incarnation, xml_text, xml_doc
 
-    def _fetch_vm_settings(self, force_update=False):
+    @staticmethod
+    def _fetch_vm_settings(wire_client):
         """
-        Issues an HTTP request (HostGAPlugin) for the vm settings and returns the response as an ExtensionsGoalStateFromVmSettings.
+        Issues an HTTP request (HostGAPlugin) for the vm settings and returns the response as an ExtensionsGoalState.
         """
         vm_settings, vm_settings_updated = (None, False)
 
         if conf.get_enable_fast_track():
             try:
-                vm_settings, vm_settings_updated = self._wire_client.get_host_plugin().fetch_vm_settings(force_update=force_update)
+                try:
+                    vm_settings, vm_settings_updated = wire_client.get_host_plugin().fetch_vm_settings()
+                except ResourceGoneError:
+                    # retry after refreshing the HostGAPlugin
+                    GoalState.update_host_plugin_headers(wire_client)
+                    vm_settings, vm_settings_updated = wire_client.get_host_plugin().fetch_vm_settings()
```

Comment:
We also need to handle parse errors from retries on 410, so now the retry is in an inner try/except